mern-ecommerce/controllers/user.js

```diff
@@ -40,6 +41,20 @@ exports.update = (req, res) => {
   );
 };
 
+exports.productById = (req, res, next, id) => { 
+  Product.findById(id) 
```

Comment:
The model being used is `Product` but it's not imported. It should be imported at the top of the file along with other models. Add `const { Product } = require('../models/product');` at the top of the file.